File: JavaScript/8js 3/whoseMove.js
/*
|--------------------------------------------------------------------------
| Simple Fun #261: Whose Move
|--------------------------------------------------------------------------
|
*/

function whoseMove(lastPlayer, win) {
   return lastPlayer == "black" && win == false || lastPlayer == "white" && win == true ? "white" : "black"
}


const whoseMove = (l, w) => ['black', 'white'][(+w - (l == 'black' ? -1 : 0)) % 2];


function whoseMove(lastPlayer, win) {
   return win ? lastPlayer : lastPlayer == "white" ? "black" : "white"
}


function whoseMove(lastPlayer, win) {
   return lastPlayer && win ? lastPlayer : {
      'black': 'white',
      'white': 'black'
   } [lastPlayer];
}


function whoseMove(lastPlayer, win) {
   return lastPlayer === `black` ^ win ? `white` : `black`;
}


console.log(whoseMove("black", false), "white")
console.log(whoseMove("white", true), "white")
console.log(whoseMove("white", false), "black")

File: Python/8py 3/whoseMove.py
"""
|--------------------------------------------------------------------------
| Simple Fun #261: Whose Move
|--------------------------------------------------------------------------
|
"""

def whoseMove(lastPlayer, win):
   return "white" if lastPlayer == "black" and win == False or lastPlayer == "white" and win == True else "black"


def whoseMove(lastPlayer, win):
   return ["white", "black"][(lastPlayer == "black") == win]


def whoseMove(lastPlayer, win):
   return lastPlayer if win else 'white' if lastPlayer == 'black' else 'black'


def whoseMove(lastPlayer, win):
   return {"white", "black"}.difference({lastPlayer}).pop() if not win else lastPlayer


print(whoseMove('black', False),'white')
print(whoseMove('white', False),'black')
print(whoseMove('black', True),'black')
print(whoseMove('white', True),'white')